docs/public/operator-hub.adoc

```diff
@@ -1,271 +1,136 @@
 
-To start an `EtcdCluster` service from `etcdoperator.v0.9.4` Operator, you need
-to execute:
+=== Listing all available Operators
 
-[source,shell]
+Before deploying your first Operator, have a look at what is available:
+[source,sh]
 ----
-$ odo service create etcdoperator.v0.9.4 --crd EtcdCluster
+  $ odo catalog list services
+  Operators available in the cluster
+  NAME                          CRDs
+  etcdoperator.v0.9.4           EtcdCluster, EtcdBackup, EtcdRestore
 ----
 
-This is exactly same command as that shown in <<dry-run,printing the YAML of
-service>> section above but without the `--dry-run` flag.
+In above output, `etcdoperator.v0.9.4` is the Operator while `EtcdCluster`,
+`EtcdBackup` and `EtcdRestore` are the CRDs provided by this Operator.
 
-At the moment, `odo` is unable to list services started from an Operator. To
-check if the above command succeeded in starting a service, use `kubectl` or
-`oc`:
 
-[source,shell]
-----
-$ kubectl get EtcdCluster
-----
-
-At the time of writing this document, above command worked out of the box
-because, as mentioned in the note, `etcdoperator.v0.9.4` 's definition has a
-valid example for `EtcdCluster` embedded into it. This can be checked by doing:
+=== Deploying etcd to a cluster
 
+In this example, we will be deploying our link:https://operatorhub.io/operator/etcd[installed etcd Operator] to an OpenShift / Kubernetes cluster.
 
-[source,shell]
+. Create a etcd service from the `etcdoperator.v0.9.4` Operator:
++
+[source,sh]
 ----
-$ kubectl get csv/etcdoperator.v0.9.4 -o yaml
+  $ odo service create etcdoperator.v0.9.4 --crd EtcdCluster
 ----
 
-and referring to the `alm-examples` section under `annotations` in the
-`metadata` of the `etcdoperator.v0.9.4` Operator.
-
-If you're using OpenShift, you can replace `kubectl` with `oc` in above
-command.
-
-If there is placeholder/invalid data or no data in the aforementioned section
-of the Operator's definition, `odo` won't be able to start the service. As an
-example, refer to the YAML definition of `EtcdBackup` in the
-`etcdoperator.v0.9.4` 's `metadata`:
-
-[source,yaml]
+. Confirm the Operator backed service was deployed:
++
+[source,sh]
 ----
-apiVersion: etcd.database.coreos.com/v1beta2
-kind: EtcdBackup
-metadata:
-  name: example-etcd-cluster-backup
-spec:
-  etcdEndpoints:
-  - <etcd-cluster-endpoints>
-  s3:
-    awsSecret: <aws-secret>
-    path: <full-s3-path>
-  storageType: S3
+  $ odo service list
 ----
 
-Here we can see some placeholder data in the form of `<etcd-cluster-endpoints>`
-, `<aws-secret>` and `<full-s3-path>` that the user is expected to set to
-appropriate value for the service to start.
+It is important to note that `EtcdBackup` and `EtcdRestore` are **not** available for deployment as they require interactive parameters.
```

Comment:
This is not entirely correct. I mean, one can't do `odo service create etcdoperator.v0.9.4 --crd EtcdBackup` or `odo service create etcdoperator.v0.9.4 --crd EtcdRestore` and expect it to work OOTB. They must redirect its alm-example to a yaml, modify the yaml to contain correct values and then use the `--dry-run` flag.

But, it's not worth holding this PR for that reason since `--dry-run` is properly documented already.

Reply:
It's not entirely correct, but still a valid statement that they don't work out of the box. I don't think we should be encouraging users to have to modify a YAML, copy over `metadata` annotations and then try again.

Reply:
I agree that we shouldn't be encouraging yaml modificatoin, but we're allowing them to pipe and edit a yaml while we get around to working on #2799 and #2785.